packages/reactDom/src/ReactServerDOM.ml

```diff
@@ -404,14 +435,26 @@ let rec to_html ~fiber (element : React.element) : (Html.element * json) Lwt.t =
 
 and render_lower_case ~fiber ~key ~tag ~attributes ~children =
   let children = ReactDOM.moveDangerouslyInnerHtmlAsChildren attributes children in
-
+  let context = Fiber.get_context fiber in
+  let push_chunk id chunk =
+    match chunk with
+    | Model.Chunk_value json ->
+        context.index <- id;
+        context.push (client_value_chunk_script id json)
+    | Model.Chunk_component_ref json ->
+        context.index <- id;
+        context.push (client_reference_chunk_script id json)
+  in
+  let context : Model.stream_context =
+    { push = push_chunk; close = context.close; chunk_id = context.index; pending = 0 }
```

Comment:
why creating a new context and not passing it via props?

Reply:
We discussed this on a call and agreed that we may need to improve the fiber and model contexts.
For now, this context will be as follows because we need to push to the chunk of the action.